pandas/tests/io/sas/test_sas7bdat.py

```diff
@@ -183,6 +183,18 @@ def test_date_time(datapath):
     tm.assert_frame_equal(df, df0)
 
 
+def test_compact_numerical_values(datapath):
+    # Regression test for #21616
+    fname = datapath("io", "sas", "data", "cars.sas7bdat")
+    df = pd.read_sas(fname, encoding='latin-1')
+    fname = datapath("io", "sas", "data", "cars.csv")
+    df0 = pd.read_csv(fname)
+    # The two columns CYL and WGT have column
+    # width < 8 in cars.sas7bdat
+    assert (df['WGT'] == df0['WGT']).all()
```

Comment:
Should use `tm.assert_frame_equal` here instead

Reply:
Changed to tm.assert_series_equal now.